providers/packetframe/api.go

```diff
@@ -0,0 +1,188 @@
+package packetframe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+const (
+	mediaType      = "application/json"
+	defaultBaseURL = "https://packetframe.com/api/"
+)
+
+func (c *packetframeProvider) fetchDomainList() error {
+	c.domainIndex = map[string]zone{}
+	dr := &domainResponse{}
+	endpoint := "dns/zones"
+	if err := c.get(endpoint, dr); err != nil {
+		return fmt.Errorf("failed fetching domain list (Packetframe): %s", err)
```

Comment:
When using fmt.Errorf, use %w instead of %s for errors. (this wraps err in the newly-created error for better diagnostic output)